hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/MergeOnReadSnapshotRelation.scala

```diff
@@ -45,13 +45,27 @@ case class MergeOnReadSnapshotRelation(override val sqlContext: SQLContext,
                                        private val globPaths: Seq[Path],
                                        private val userSchema: Option[StructType],
                                        private val prunedDataSchema: Option[StructType] = None)
-  extends BaseMergeOnReadSnapshotRelation(sqlContext, optParams, metaClient, globPaths, userSchema, prunedDataSchema) {
+  extends BaseMergeOnReadSnapshotRelation(sqlContext, optParams, metaClient, globPaths, userSchema, prunedDataSchema) with SparkAdapterSupport {
 
   override type Relation = MergeOnReadSnapshotRelation
 
   override def updatePrunedDataSchema(prunedSchema: StructType): MergeOnReadSnapshotRelation =
     this.copy(prunedDataSchema = Some(prunedSchema))
 
+  def toHadoopFsRelation: HadoopFsRelation = {
+    fileIndex.shouldBroadcast = true
+    HadoopFsRelation(
+      location = fileIndex,
+      partitionSchema = fileIndex.partitionSchema,
+      dataSchema = fileIndex.dataSchema,
+      bucketSpec = None,
+      fileFormat = sparkAdapter.createMORBootstrapFileFormat(shouldExtractPartitionValuesFromPartitionPath,
+        sparkSession.sparkContext.broadcast(tableState),
+        sparkSession.sparkContext.broadcast(HoodieTableSchema(tableStructSchema, tableAvroSchema.toString, internalSchemaOpt)),
+        metaClient.getTableConfig.getTableName, mergeType, mandatoryFields, isMOR = true, isBootstrap = false).get,
+      optParams)(sparkSession)
+  }
```

Comment:
Could the `HadoopFsRelation` object be directly created in `DefaultSource` without going through `MergeOnReadSnapshotRelation`?  (I have a similar question on bootstrap query too)

Reply:
I did that here:
https://github.com/jonvex/hudi/blob/c11c20b1110dbf0c5d5c4be98d2362b5648d02ae/hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieFormatUtils.scala
All the code is copied from the various relation classes. We could do something like this if you think it's better this way

Reply:
Yeah, sg